bundlerepository/src/main/java/org/apache/felix/bundlerepository/impl/SpecXMLPullParser.java

```diff
@@ -328,6 +330,11 @@ private static Requirement parseRequirement(XmlPullParser reader) throws Excepti
         parseAttributesDirectives(reader, attributes, directives, REQUIREMENT);
         requirement.setAttributes(attributes);
 
+        String effective = directives.get("effective");
+        if (effective != null && !effective.equals("resolve")) {
+        	return null;
```

Comment:
Why return null here? It's still a valid requirement but just not a resolve requirement. Did this cause any issues anywhere?